### DISCUSSION TITLE
Question: What place has .NET Native after Project Reunion

### DISCUSSION THREAD

**Discussion by pjmlp** [microsoft/WindowsAppSDK#375] [Q&A]
Apparently the long term roadmap will be to merge Win32 and UWP worlds into a common development experience.

Given the newly introduced C#/WinRT and the ONE .NET effort announced at BUILD, where does it leave .NET Native in a Project Reunion world?

**Comment by mveril** [comment]
For me .NET Native should be deleted to be replaced by .NET 5

**Comment by Gavin-Williams** [comment]
The only way .Net Native should survive is if it's available to .Net 5. Are we going to reunite or what? Or are we going to still be in the same situation in 2 years time as we are now, using all these different frameworks and all complaining about how support is lacking for the particular variant we are stuck on. Lets get this sorted ..

.Net 5 and choose JIT or AOT at config. done.

**Comment by prayaas-a** [comment]
AOT/precompilation support should be opt-in and made available in v5.
.NET teams have worked on that in numerous projects now including CoreRT and WebAssembly. I hope it makes it to .NET 5.x because the performance boost can be huge.
In the CoreRT repo, jkotas said in early 2019:

> We definitely do want to have a great full AOT compilation story as one of the officially supported deployment options for .NET Core eventually. It is not happening for .NET Core 3. We have not really started planning for .NET Core 4.

Now obviously .NET Core 4 materialized into .NET 5 but this makes me think it's on the burner for sooner or later.

**Comment by saint4eva** [comment]
> Now obviously .NET Core 4 materialized into .NET 5 but this makes me think it's on the burner for sooner or later.

The pandemic interfered with the roadmap, so it would likely be in .NET 6

**Comment by stevewri** [comment]
Does #105 address this question?

**Reply by HppZ** [reply]
when official support？

**Comment by HppZ** [comment]
when will .NET 5 official support arrive？MS come on…

**Comment by ccarlo88** [comment]
I think this address the question:
https://github.com/dotnet/runtimelab/issues/1120

As mentioned, NativeAOT is just experimental and it has no clear roadmap and probably no future. What I can recommend is moving on, I can recommend using Vala+GTK+Glade (for xml UI), it is very similar to C#. It converts the code to C and compile, so you even have the benefit of cross-compile and no JIT. There's quite a good amount of software made with it.
